llvm/lib/SYCLLowerIR/CleanupSYCLMetadata.cpp

```diff
@@ -34,6 +33,13 @@ void cleanupSYCLCompilerMetadata(const Module &M, llvm::StringRef MD) {
   Node->eraseFromParent();
 }
 
+void cleanupSYCLCompilerFunctionMetadata(Function &F,
+                                         llvm::StringRef MD) {
+  if (!F.getMetadata(MD))
+    return;
+  F.setMetadata(MD, nullptr);
```

Comment:
Actually, early exit is not necessary.
```suggestion
  if (F.getMetadata(MD))
    F.setMetadata(MD, nullptr);
```

Reply:
Well, originally I was deleting more metadata here. But then I found out, that actually srcloc is the only one which I should remove (now).